### PR TITLE
CHANGELOG and UpgradeGuide Updates

### PR DESCRIPTION
**Pull request by Zuhairahmed**
## Description

- Add AWS Serverless Private Endpoints + Cluster Termination Protection in v1.6.0
- Remove mention of AWS Serverless Private Endpoints in v1.5.0

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code

### CODE REVIEW

**Review by martinstibbe — APPROVED** [review]
LGTM

**Review by martinstibbe — COMMENTED** [review]
@Zuhairahmed Would we want boilerplate text to be New Features vs ENHANCEMENTS ?

**Comment by Zuhairahmed** [comment]
thanks, updated with "NEW FEATURES" to simplify. 

also will drop in comments in automation ticket for CHANGELOG, but i think we can have 4 main buckets in future: "NEW FEATURES" (which we can pick up via human adding label to PR during review), "BUG FIXES" (which we can pick up via human adding label to PR during review), "MERGED PULL REQUESTS" (these are minor dependabot or doc edits that have already been merged into master since last release), and "CLOSED ISSUES" (these are simply all the issues that have been closed since last release)

**Review by martinstibbe — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
+## [v1.6.0](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.6.0) (2022-11-16)
+
+[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.5.0...v1.6.0)
+
+**NEW FEATURES:**
```
Comment (themantissa) [inline CHANGELOG.md:5]: Not blocking but last time we used ENHANCEMENTS and now it's NEW FEATURES - we need a defined format and a script to generate so we have consistency in the CHANGELOG so it's easy to read.

**Review by themantissa — APPROVED** [review]

**Pull request merged**